### PR TITLE
FIX: Create single fmap_select_std node for use with all ANAT estimators

### LINKED ISSUE #2687: Duplicate node name "fmap_select_std" found
### What happened?

I get:

`OSError: Duplicate node name "fmap_select_std" found.`

This is single-subject. The workdir is empty, and this subject is freshly downloaded and never been run before - there is no existing derivatives directory.

### What command did you use?

```shell
/usr/bin/singularity run \
    --contain \
    --cleanenv \
    -B /tmp -B /data -B /data1 -B /data2 -B /data3 -B /n -B /cm/shared \
    /cm/shared/singularity/images/fmriprep-21.0.0.simg \
    /data/belleau_lab/E21774098 \
    /data/belleau_lab/E21774098/derivatives \
    participant \
    --fs-license-file /cm/shared/freesurfer-6.0.1/license.txt \
    --participant_label KEB08 \
    --output-spaces MNI152NLin6Asym:res-2 anat fsaverage \
    --n_cpus 4 \
    --mem-mb 16384 \
    --notrack \
    --use-aroma \
    --use-syn-sdc \
    --return-all-components \
    -vv \
    -w  /data/fmriprep-workdir/ebelleau
```

### What version of fMRIPrep are you running?

21.0.0

### How are you running fMRIPrep?

Singularity

### Is your data BIDS valid?

Yes

### Are you reusing any previously computed results?

No

### Please copy and paste any relevant log output.

```shell
last line of stdout:

220118-19:09:13,326 nipype.workflow INFO:
	 Setting-up fieldmap "auto_00001" (EstimatorType.ANAT) with <sub-KEB08_T1w.nii.gz, sub-KEB08_task-EMIST_run-01_bold.nii.gz, sub-KEB08_task-EMIST_run-02_bold.nii.gz, sub-KEB08_task-HMIST_run-01_bold.nii.gz, sub-KEB08_task-HMIST_run-02_bold.nii.gz, sub-KEB08_task-VOCBASELINEPOST_run-02_bold.nii.gz, sub-KEB08_task-VOCBASELINEPOST_run-04_bold.nii.gz, sub-KEB08_task-VOCBASELINEPRE_run-02_bold.nii.gz, sub-KEB08_task-VOCBASELINEPRE_run-04_bold.nii.gz, sub-KEB08_task-VOCMIXEDPOST_run-01_bold.nii.gz, sub-KEB08_task-VOCMIXEDPOST_run-03_bold.nii.gz, sub-KEB08_task-VOCMIXEDPRE_run-01_bold.nii.gz, sub-KEB08_task-VOCMIXEDPRE_run-03_bold.nii.gz>


stderr:

/opt/conda/lib/python3.8/site-packages/bids/config.py:39: FutureWarning: Setting 'extension_initial_dot' will be removed in pybids 0.16.
  warnings.warn("Setting 'extension_initial_dot' will be removed in pybids 0.16.",
Process Process-2:
Traceback (most recent call last):
  File "/opt/conda/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/opt/conda/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/conda/lib/python3.8/site-packages/fmriprep/cli/workflow.py", line 118, in build_workflow
    retval["workflow"] = init_fmriprep_wf()
  File "/opt/conda/lib/python3.8/site-packages/fmriprep/workflows/base.py", line 85, in init_fmriprep_wf
    single_subject_wf = init_single_subject_wf(subject_id)
  File "/opt/conda/lib/python3.8/site-packages/fmriprep/workflows/base.py", line 480, in init_single_subject_wf
    workflow.connect([
  File "/opt/conda/lib/python3.8/site-packages/nipype/pipeline/engine/workflows.py", line 161, in connect
    self._check_nodes(newnodes)
  File "/opt/conda/lib/python3.8/site-packages/nipype/pipeline/engine/workflows.py", line 769, in _check_nodes
    raise IOError('Duplicate node name "%s" found.' % node.name)
OSError: Duplicate node name "fmap_select_std" found.
```

### Additional information / screenshots

_No response_

### PR DESCRIPTION
**Pull request by effigies**
## Changes proposed in this pull request

Moves `fmap_select_std` out of a loop, as it can be duplicated unnecessarily and cause a name collision if multiple ANAT estimators are found.

Fixes #2687.

### CODE REVIEW

File: fmriprep/workflows/base.py [hunk 1]
```diff
@@ -429,7 +429,25 @@ def init_single_subject_wf(subject_id):
...
+    # Step 3: Manually connect PEPOLAR and ANAT workflows
+
+    # Select "MNI152NLin2009cAsym" from standard references.
+    # This node may be used by multiple ANAT estimators, so define outside loop.
+    from niworkflows.interfaces.utility import KeySelect
+    fmap_select_std = pe.Node(
+        KeySelect(fields=["std2anat_xfm"], key="MNI152NLin2009cAsym"),
+        name="fmap_select_std",
+        run_without_submitting=True,
+    )
```
Comment (mgxd) [inline fmriprep/workflows/base.py:441]: can this be inserted into the `if` statement? or does something else depend on this being defined?

**Review by mgxd — COMMENTED** [review]
good catch - one minor comment

File: fmriprep/workflows/base.py (hunk 1 above)
Comment (effigies) [reply fmriprep/workflows/base.py:441]: I thought about this, but then considered that somebody working further down the code might see "this variable is assumed in this loop" and miss that it's only in one condition branch within the loop. Maybe a `NameError` is at least an earlier failure than no input, though...

I can go either way, if you have a strong preference.

Comment (mgxd) [reply fmriprep/workflows/base.py:441]: I'm fine with that train of thought

**Pull request merged**